docs/docsite/rst/playbooks_variables.rst

```diff
@@ -782,21 +805,25 @@ Example::
 
     ansible-playbook release.yml --extra-vars "hosts=vipers user=starbuck"
 
-As of Ansible 1.2, you can also pass in extra vars as quoted JSON, like so::
+Escaping quotes and other special characters:
 
-    --extra-vars '{"pacman":"mrs","ghosts":["inky","pinky","clyde","sue"]}'
+If your variable definition string contains quotes (e.g. JSON), and especially if the 
+your variables' themselves contain quotes, you'll need to ensure you're escaping these 
+quotes appropriately for both your data format (JSON or YAML), and for the shell.  This 
+can lead to some very difficult to read definitions::
 
-The ``key=value`` form is obviously simpler, but it's there if you need it!
+    ansible-playbook arcade.yml --extra-vars "{\"name\":\"Conan O\'Brien\"}"
+    ansible-playbook arcade.yml --extra-vars '{"name":"Conan O'\\\''Brien"}'
+    ansible-playbook script.yml --extra-vars "{\"dialog\":\"He said \\\"I just can\'t get enough of those single and double-quotes"\!"\\\"\"}"
+
+In these cases, it's probably best to use a JSON or YAML file contianing the variable 
+definitions.
 
 .. note:: Values passed in using the ``key=value`` syntax are interpreted as strings.
           Use the JSON format if you need to pass in anything that shouldn't be a string (Booleans, integers, floats, lists etc).
-
-As of Ansible 1.3, extra vars can be loaded from a JSON file with the ``@`` syntax::
-
-    --extra-vars "@some_file.json"
-
-Also as of Ansible 1.3, extra vars can be formatted as YAML, either on the command line
-or in a file as above.
+.. note:: JSON format isn't available prior to Ansible 1.2
+.. note:: YAML format isn't available prior to Ansible 1.3
+.. note:: @filename isn't available prior to Ansible 1.3
```

Comment:
we normally use `version added` entries when we first introduce the issues as most people don't scroll down to the notes. Also we 'mask' any version added that is too old, which is not possible with the notes.

Reply:
I'm trying to find an example of this to emulate.  I've found lots of `version_added: 2.1` lines in `.py` files, but I haven't yet found the `.rst` equivalent.  I'll keep looking, though.

Reply:
The closest strings I found in ansible/docs/docsite/rst were:
`In {{ version_number }} we added` at the beginning of 4 lines, and
`added in {{ version_number }}` at various places within 3 lines.

```
playbooks_blocks.rst:In 2.0 we added a block feature to allow for logical grouping of tasks and even
playbooks_debugger.rst:In 2.1 we added a ``debug`` strategy. This strategy enables you to invoke a debugger when a task has
playbooks_strategies.rst:In 2.0 we added a new way to control play execution, ``strategy``, by default plays will
porting_guide_2.4.rst:  - In 2.4.1 we added a toggle to allow you to control this behaviour, 'top' will be the pre 2.4, 'bottom' will use the current playbook hosting the task and 'all' will use them all from top to bottom.
playbooks_intro.rst:    The ``remote_user`` parameter for tasks was added in 1.4.
playbooks_reuse_roles.rst:    - name: added in 2.4, previouslly you used 'include'
playbooks_reuse.rst:Includes and imports (added in 2.4) allow users to break up large playbooks into smaller files, which can be used across multiple parent playbooks or even multiple times within the same Playbook.
```

Do you have a preferred format?  Neither one looks like it would trigger any automated masking (though I might not be giving the mask function enough credit).  Is the masking a `.py`-documentation-only thing, or does it apply to the `.rst` files as well?

Reply:
in the rst:
```
.. versionadded:: 2.0
```